From 45f198113a1807faf094e3529b03da7622acc3ed Mon Sep 17 00:00:00 2001
From: Jens Granseuer <jensgr@gmx.net>
Date: Wed, 20 May 2009 21:20:04 +0000
Subject: [common] don't unref NULL GConf entries

 2009-05-03  Jens Granseuer  <jensgr@gmx.net>
 
 	Patch by: <yselkowitz@users.sourceforge.net>
diff --git a/capplets/common/gconf-property-editor.c b/capplets/common/gconf-property-editor.c
index 518fd9f..31f9e0e 100644
--- a/capplets/common/gconf-property-editor.c
+++ b/capplets/common/gconf-property-editor.c
@@ -351,7 +351,8 @@ gconf_peditor_new (const gchar           *key,
 	gconf_entry = gconf_client_get_entry (client, GCONF_PROPERTY_EDITOR (obj)->p->key, NULL, TRUE, NULL);
 	GCONF_PROPERTY_EDITOR (obj)->p->callback (client, 0, gconf_entry, obj);
 	GCONF_PROPERTY_EDITOR (obj)->p->inited = TRUE;
-	gconf_entry_free (gconf_entry);
+	if (gconf_entry)
+		gconf_entry_free (gconf_entry);
 	g_object_unref (client);
 
 	return obj;
--
cgit v0.8.2
